### PR TITLE
Add more feature to the todo app

### PR DESCRIPTION
**Pull request by IshmaelKargbo**
In this branch I worked on:

1. Add todo to the store
2. Toggle the todo state from edit to normal or normal to edit
3. Edit and update the previous todo that is been edited
4. Remove a todo from the store

Please review and give your comments and feedback
Thanks 😄

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -4,32 +4,94 @@ import TodoStore from './modules/todo-store.js';
...
+        renderTodo();
+      });
+    }
   });
 };
 
+// add new todo to the store
 const addTodo = (e) => {
   e.preventDefault();
   const todo = Object.fromEntries(new FormData(e.target).entries());
```
Comment (brytebee) [inline src/index.js:97]: - [ ] **Required:** Great job so far. However, I have observed that 
- the app does not add items to the list.
- the localStorage key is never created.
- Below is a gif for more.


app_misbahave

Kindly fix the app behavior and request another review.

File: src/modules/todo-store.js [hunk 2]
```diff
@@ -31,6 +31,39 @@ class TodoStore {
     this.#store.push(todo);
     this.#backup();
   }
+
```
Comment (brytebee) [inline src/modules/todo-store.js:34]: - [ ] **Optional:** There are also console errors that you should definitely consider fixing.

image

https://github.com/IshmaelKargbo/todo-list/blob/ca93a29b5aecca7131d2a95ad56f510f4f61d584/src/modules/todo-store.js#L19-L22

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @IshmaelKargbo👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- Linters are good and green.✔️
- Good PR summary.✔️
- Proper GitHub flow used.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (IshmaelKargbo) [reply src/index.js:97]: Thanks

File: src/modules/todo-store.js (hunk 2 above)
Comment (IshmaelKargbo) [reply src/modules/todo-store.js:36]: thanks

File: src/index.js (hunk 1 above)
Comment (IshmaelKargbo) [reply src/index.js:97]: By the look at the gif, it seems like maybe you are not running the latest version of the branch

**Review by elyor-doniyorov — APPROVED** [review]
Hi @IshmaelKargbo, 

In my opinion, your project is rcomplete!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**